manager/src/main/java/com/redhat/service/smartevents/manager/api/user/validators/processors/CloudProviderConstraintValidator.java

```diff
@@ -0,0 +1,80 @@
+package com.redhat.service.smartevents.manager.api.user.validators.processors;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.validation.ConstraintValidatorContext;
+
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.InvalidCloudProviderException;
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.InvalidRegionException;
+import com.redhat.service.smartevents.manager.api.models.requests.BridgeRequest;
+import com.redhat.service.smartevents.manager.dao.CloudProviderDAO;
+import com.redhat.service.smartevents.manager.models.CloudProvider;
+import com.redhat.service.smartevents.manager.models.CloudRegion;
+import org.apache.commons.lang3.StringUtils;
+
+@ApplicationScoped
+public class CloudProviderConstraintValidator extends BaseConstraintValidator<ValidCloudProvider, BridgeRequest> {
+
+    static final String ID_PARAM = "id";
+
+    static final String REGION_PARAM = "region";
+
+    static final String CLOUD_PROVIDER_NOT_VALID = "Cloud Provider '{id}' is not valid.";
+
+    static final String CLOUD_PROVIDER_NOT_ENABLED = "Cloud Provider '{id}' is not enabled.";
+
+    static final String CLOUD_REGION_NOT_VALID = "The Region '{region}' for Cloud Provider '{id}' is not valid.";
+
+    static final String CLOUD_REGION_NOT_ENABLED = "The Region '{region}' for Cloud Provider '{id}' is not enabled.";
+
+    @Inject
+    CloudProviderDAO cloudProviderDAO;
+
+    @Override
+    public boolean isValid(BridgeRequest bridgeRequest, ConstraintValidatorContext context) {
+
+        /*
+            Not valid to have a null or empty cloud_provider or region field, but validation failure
```

Comment:
Indeed, that does remove the "double validation" problem reported by @sutaakar however the `ConstraintViolation` thrown by the Java Bean Validation Framework will not have a dynamic payload and hence cannot be mapped to the specific `BridgeError` representing an invalid Cloud Provider parameter. I suspect you'll get the generic `UnclassifiedConstraintViolationException`, `OPENBRIDGE-21`.

IMO you'd be better off removing the validation annotation from `BridgeRequest` and keep it all in here.

Reply:
`Name` and "other parameters" is already an issue. Something @kelvah keeps reminding us about.

"Other parameters" are validated in a `ConstraintValidator` and set the dynamic payload.

"Name" is validated programmatically within our service after "other parameters" are valid.

So, Users have a "two phase" validation of both `Bridge` and `Processor` at the moment.
- Java Bean Validation constraints. When these are valid...
- Our programmatic constraints. When these are valid... 
- The request is valid.

Reply:
@robpblake yes it is one of the spinning plates. 

In response to this thread @kelvah is writing a document about validation and (I assume) will add to a ticket.

It definitely is not something to be fixed on this PR; but you still might want to consider removing use of `@NotEmpty` that will, I suspect, map to the generic `UnclassifiedConstraintViolationException` (since we're unable to set it's dynamic payload...)